cpp/ql/lib/ext/CAtlList.model.yml

```diff
@@ -0,0 +1,25 @@
+extensions:
+  - addsTo:
+      pack: codeql/cpp-all
+      extensible: summaryModel
+    data: # namespace, type, subtypes, name, signature, ext, input, output, kind, provenance
+      - ["", "CAtlList", True, "AddHead", "", "", "Argument[*@0]", "Argument[-1].Element[@]", "value", "manual"]
+      - ["", "CAtlList", True, "AddHeadList", "", "", "Argument[*0].Element[@]", "Argument[-1].Element[@]", "value", "manual"]
+      - ["", "CAtlList", True, "AddTail", "", "", "Argument[*@0]", "Argument[-1].Element[@]", "value", "manual"]
+      - ["", "CAtlList", True, "AddTailList", "", "", "Argument[*0].Element[@]", "Argument[-1].Element[@]", "value", "manual"]
+      - ["", "CAtlList", True, "FindIndex", "", "", "Argument[0]", "ReturnValue", "taint", "manual"]
```

Comment:
Oh, yes I agree. We should not have taint on positions. I added this at some point and then changed my mind as we should do the proper thing (this is related to the "Lots of container related models contains code allows you to do stuff like" section of the PR description).

I'll remove the taint on positions. I totally agree this shouldn't be there.

Reply:
Fixed in 257c657608cb6759a4d8e5ff287d9bfb17f13368. I kept the second `GetAt` model since that returns a reference to the actual element and not a `POSITION`. A reference to the actual value should be tainted (and a subsequent lvalue-to-rvalue conversion on the reference will also be a taint-step, which will provide taint-flow from the return value of `GetAt` to the `GetAt` expression).

Reply:
Fixed in de75e033beb7b871e23f7dc706fd904bcbf19529